include/psen_scan/reply_msg.h

```diff
@@ -142,7 +143,7 @@ inline ReplyMsg::RawType ReplyMsg::toCharArray()
 
   // TODO check limits
   std::string data_str(os.str());
-  // TODO check if lengths match
+  assert(data_str.length() == REPLY_MSG_SIZE);
```

Comment:
I think it would be good to add a description to assertions. I think it will work like this, but you'll have to double-check it ...
```suggestion
  assert(("Message data has not the expected size", data_str.length() == REPLY_MSG_SIZE));
```